src/main/java/mate/academy/dao/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,47 @@
+package mate.academy.dao.impl;
+
+import mate.academy.dao.ManufacturerDao;
+import mate.academy.db.Storage;
+import mate.academy.lib.Dao;
+import mate.academy.model.Manufacturer;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufactures(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        return Storage.manufacturers.stream()
+                .filter(p -> p.getId().equals(id))
+                .findFirst();
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Optional<Manufacturer> first = Storage.manufacturers.stream()
+                .filter(p -> p.getId().equals(manufacturer.getId()))
+                .findFirst();
+        first.get().setName(manufacturer.getName());
+        first.get().setCountry(manufacturer.getCountry());
+        return first.get();
```

Comment:
```suggestion
       IntStream.range(0, Storage.manufacturers.size())
                .filter(i -> Storage.manufacturers.get(i).getId().equals(manufacturer.getId()))
                .forEach(i -> Storage.manufacturers.set(i, manufacturer));
        return manufacturer;
```
this way you won't need to modify this code if manufacturers' fields changes over time

Reply:
fix